JamesRobertSutcliffe/personality-music-recommender
# Cache user email

Waiting on #60 

This PR introduces a solution to share the fetched email across routes and components without the need to pass it as a prop or parameter during redirection. Initially, I considered using cookies or localStorage, but these options pose security and privacy risks due to XSS attacks.

Previously, only the Personalised Homepage had access to the user details. However, we needed a way to make this information available to other routes. This is where the context API comes in. With the implementation of the context API, all routes and components should now have access to the user's email, which serves as a unique identifier.

This enhancement allows us to retrieve user-specific data from the database in all of our components and pages.

## Testing
1. Sign in. You will get redirected to `/test-page`
1. Ensure you able to access the updated value of `userEmail` in `/test-page`
1. `npm run test`
